src/types/Firestore.ts

```diff
@@ -1,88 +1,94 @@
-import { Venue } from "./Venue";
-import { Purchase } from "./Purchase";
-import { User } from "./User";
+import { WithId } from "utils/id";
+import { AdminRole } from "hooks/roles";
+
 import {
   RestrictedChatMessage,
   PrivateChatMessage,
 } from "components/context/ChatContext";
-import { VenueEvent } from "./VenueEvent";
-import { Table } from "./Table";
-import { PartyMapVenue } from "./PartyMapVenue";
 import { Reaction } from "components/context/ExperienceContext";
-import { WithId } from "utils/id";
+
 import { CampVenue } from "./CampVenue";
 import { ChatRequest } from "./ChatRequest";
+import { PartyMapVenue } from "./PartyMapVenue";
+import { Purchase } from "./Purchase";
 import { Role } from "./Role";
-import { AdminRole } from "hooks/roles";
+import { Table } from "./Table";
+import { User } from "./User";
+import { Venue } from "./Venue";
+import { VenueEvent } from "./VenueEvent";
 
-interface VenueStatus {
-  currentVenue: boolean;
-  currentEvent: boolean;
-  eventPurchase: boolean;
-  venueChats: boolean;
-  venueEvents: boolean;
-  userPurchaseHistory: boolean;
-}
+export type AnyVenue = Venue | PartyMapVenue | CampVenue;
 
 interface Experience {
   reactions: Record<string, Reaction>;
   tables: Record<string, Table>;
 }
 
-type VenueTimestamps = Record<keyof VenueStatus, number>;
-export type AnyVenue = Venue | PartyMapVenue | CampVenue;
-
 interface UserVisit {
   timeSpent: number;
 }
 
+export type ValidFirestoreKeys = keyof FirestoreData | keyof FirestoreOrdered;
+
 export interface Firestore {
```

Comment:
This was de-structured to make separate types for `FirestoreStatus`, `FirestoreData`, `FirestoreOrdered`